buildSrc/src/main/kotlin/otel.publish-conventions.gradle.kts

```diff
@@ -63,3 +80,36 @@ if (System.getenv("CI") != null) {
     sign(publishing.publications["mavenPublication"])
   }
 }
+
+if (!project.name.startsWith("bom")) {
+  project.afterEvaluate {
+    val sbomName = "opentelemetry-java_" + base.archivesName.get()
+    tasks.cyclonedxBom {
+      outputName = "$sbomName.cyclonedx"
+      includeConfigs = listOf("runtimeClasspath")
+      skipConfigs = listOf("compileClasspath", "testCompileClasspath")
+      destination = project.file("build/cyclonedx")
+      outputFormat = "json"
+    }
+    spdxSbom {
+      targets {
+        // Create a target to match the published jar name.
+        // This is used for the task name (spdxSbomFor<SbomName>)
+        // and output file (<sbomName>.spdx.json).
+        create(sbomName) {
+          scm {
+            uri.set("https://github.com/" + System.getenv("GITHUB_REPOSITORY"))
+            revision.set(System.getenv("GITHUB_SHA"))
+          }
+          document {
+            name.set(sbomName)
+            namespace.set("https://opentelemetry.io/spdx/" + UUID.randomUUID())
+          }
+        }
+      }
+    }
+    tasks.named("assemble") {
```

Comment:
If I run `./gradlew publishToMavenLocal` I get errors because the publish step is looking for the SBOMs but `publishToMavenLocal` doesn't depend on `assemble`. Is there a better task we can hook into? Perhaps `javaCompile`.

Reply:
I've added an additional task dependency for the publish tasks.  I didn't want to remove the `assemble` one because we still want it to execute in the `build` workflow.  I also didn't want to hook in at too low of level to avoid running it unnecessarily on dev machines.

Reply:
Makes sense 👍